### ISSUE TITLE
Fix the problem that caused ";" to introduce 'NA's

### ISSUE THREAD

**Issue by maurolepore** [2DegreesInvesting/tiltIndicator#439]
Follow up https://github.com/2DegreesInvesting/tiltIndicator/issues/405

**Comment by maurolepore** [comment]
``` r
library(tibble)
devtools::load_all()
#> ℹ Loading tiltIndicator
packageVersion("tiltIndicator")
#> [1] '0.0.0.9075'

# Bad: What we have
companies <- tibble(
  company_id = "a",
  type = "a",
  clustered = "a",
  activity_uuid_product_uuid = "a",
  tilt_sector = "a",
  tilt_subsector = "a",
  # We want to match `sector = "b"`
  sector = "a b; c d",
  subsector = "b",
)

scenarios <- tibble(
  type = "a",
  # TODO: Consider the case when sector is NA
  # We want to match `sector = "b"`
  sector = "b",
  subsector = "b",
  scenario = "b",
  year = 2050,
  reductions = 1,
)


pstr(companies, scenarios)
#> # A tibble: 1 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 a            <NA>       <NA>             NA

# Good: What we want
companies <- tibble(
  company_id = "a",
  type = "a",
  clustered = "a",
  activity_uuid_product_uuid = "a",
  tilt_sector = "a",
  tilt_subsector = "a",
  # We want to match `sector = "b"`. 
  # * Each word separated by space " " became an item.
  # * Each word separated by space ";" became an item.
  sector = c("a", "b", "c", "d"),
  # We want to match `subsector` regardless of `sector` even if `subsector` is
  # `NA`, but that should be the focus of a separate issue
  subsector = "b",
)
companies
#> # A tibble: 4 × 8
#>   company_id type  clustered activity_uuid_product_…¹ tilt_sector tilt_subsector
#>   <chr>      <chr> <chr>     <chr>                    <chr>       <chr>         
#> 1 a          a     a         a                        a           a             
#> 2 a          a     a         a                        a           a             
#> 3 a          a     a         a                        a           a             
#> 4 a          a     a         a                        a           a             
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid
#> # ℹ 2 more variables: sector <chr>, subsector <chr>

pstr(companies, scenarios)
#> # A tibble: 3 × 4
#>   companies_id grouped_by risk_category value
#>   <chr>        <chr>      <chr>         <dbl>
#> 1 a            a_b_2050   high              1
#> 2 a            a_b_2050   medium            0
#> 3 a            a_b_2050   low               0
```

<sup>Created on 2023-06-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by Tilmon** [comment]
@maurolepore I reviewed this issue again and think we should put it on hold until Bob is done with refactoring of tiltIndicatorBefore. I am right now updating the sector mappers so that this case with ";" ideally won't happen again. Once Bob is done, we can see whether it worked and we indeed avoided ";". Otherwise, we need to think whether we want to exclude these cases again from the results or put the different sectors in multiple rows as you did in the reprex.

**Comment by maurolepore** [comment]
Closing in favor or #448
